src/main/java/ch/njol/skript/Skript.java

```diff
@@ -1349,7 +1350,7 @@ public boolean check(final @Nullable ExpressionInfo<?, ?> i) {
 	private static final Collection<SkriptEventInfo<?>> events = new ArrayList<>(50);
 
 	private static final List<SyntaxElementInfo<? extends Structure>> normalStructures = new ArrayList<>(10);
-	private static final List<SyntaxElementInfo<? extends PreloadingStructure>> preloadingStructures = new ArrayList<>(10);
+	private static final List<PreloadingStructureInfo<? extends PreloadingStructure>> preloadingStructures = new ArrayList<>(10);
```

Comment:
There should not be a difference between PreloadingStructure and Structure. Remove the former and merge its features into the normal class.
You can do this by having a default priority for 'normal' structures (I think 5 is a good number).